regression_pathology_features.py

```diff
@@ -0,0 +1,147 @@
+"""
+Run regression using a pathology characteristic as the independent variable and
+evaluation metric as the dependent variable.
+"""
+from argparse import ArgumentParser
+import pandas as pd
+
+from eval_constants import LOCALIZATION_TASKS
+from utils import run_linear_regression
+
+
+def normalize(column):
+    if column.min() == column.max():
+        return column
+    else:
+        return (column-column.min())/(column.max()-column.min())
+
+
+def format_ci(row):
+    """Format confidence interval."""
+    def format_stats_sig(p_val):
+        """Output *, **, *** based on p-value."""
+        stats_sig_level = ''
+        if p_val < 0.001:
+            stats_sig_level = '***'
+        elif p_val < 0.01:
+            stats_sig_level = '**'
+        elif p_val < 0.05:
+            stats_sig_level = '*'
+        return stats_sig_level
+
+    # CI on linear regression coefficients
+    p_val = row['coef_pval']
+    #TODO: when does feature = prob, and why *4?
```

Comment:
@alexaaag when does feature equal prob? and why do we multiply by 4?

Reply:
We multiply by 4 to get the Bonferroni corrected p value (since we did four tests). The Bonferroni correction is used to reduce the chances of obtaining false-positive results (type I errors) when multiple pair wise tests are performed on a single set of data. We didn't do the correction on the probability regression since only one test is run (on different pathologies, and thus different sets of data).

If you read the literature, they formulate it as: reject if p_val < 0.05/4. In this code I just multiply the left hand side by 4 - I think they are equivalent.

Reply:
got it! And when would `feature` ever equal `prob`?

Reply:
never in this script now we have separated regression on pathology features and probability. In the previous version feature (the independent variable) could equal prob when running regression on probability.